titus_isolate/isolate/cpu.py

```diff
@@ -4,157 +4,174 @@
 from titus_isolate.model.workload import Workload
 from titus_isolate.utils import get_logger
 
+import time
+
 from titus_optimize.compute import optimize_ip
 
 log = get_logger()
 
+class IntegerProgramCpuAllocator:
```

Comment:
It would be nice if both allocators extended an Abstract Base Class to provide some interface enforcement, description.

Reply:
done